client/components/Login.js

```diff
@@ -0,0 +1,90 @@
+import React, {Component} from 'react';
+import axios from 'axios'
+
+class Login extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: "",
+      password: "",
+      serverMsg: "",
+    };
+  }
+
+
+  // todo: make username selected after didMount
+  componentDidMount = () => { };
+
+  componentWillUnmount = () => {
+    clearTimeout(this.timeoutID)
+  };
+
+  showErrMsg = msg => {
+    if (this.timeoutID){ // clear the previous timeoutID
+      clearTimeout(this.timeoutID)
+    }
+
+    this.setState({serverMsg: msg});
+    this.timeoutID = setTimeout(() => this.setState({serverMsg: ""}), 8000)
+  };
+
+  textChangeHandler = evt => {
+    this.setState({
+      [evt.target.id]: evt.target.value || ""
+    })
+  };
+
+  onKeyDown = evt => {
+    if (this.state.username === "" || this.state.password === ""){
+      return
```

Comment:
I see a few inconsistencies about the code format, e.g. semicolon after the statement,
Have you ever used any lint tool?

Reply:
No, I let the editor(WebStorm) do all the styling suggestion.